### ISSUE TITLE
Facebook Validation problem

### ISSUE THREAD

**Issue by jcambien** [hybridauth/hybridauth#990]
Hi,

As any website running a Facebook application today, you have been faced with the re-validation requested by Facebook with the deadline of 1st August.
They're actually refusing the validation of our application because HybridAuth makes redirections between the button click and the Facebook login screen.
Any idea/solution to makes HybridAuth giving authentication link and avoiding those redirections ?
We're currently running HybridAuth 3.0

Thank you

**Comment by StorytellerCZ** [comment]
Hi @jcambien ,
I'm not sure I understand what you are asking for. Redirection is the recommended process. Another option is pop-up. Redirection is fine, you just need to whitelist the url to where the user will be redirected after they successfully confirm login to Facebook.

**Comment by jcambien** [comment]
@StorytellerCZ Facebook is recommending us to avoid any redirection between the moment the user clicks the "Connect with Facebook" and the Facebook login screen, otherwise they refuse to validate our application. But as far as I know, Hybridauth makes the redirection to the login screen himself in the `authenticate()` method. The solution could be replacing the Facebook login button destination with a direct link to Facebook, but i'm scared it'll break internal Hybridauth logic.

**Comment by ApacheEx** [comment]
@jcambien, can you give a link where this problem is described? 
I'm wondering because it makes sense for client side implementation (no HA needed). But for server side, we have to make a redirection.

**Comment by jcambien** [comment]
Sorry no link to provide because it's in the validation process which is private.
This is the given reason :

![fb-validation](https://user-images.githubusercontent.com/153502/44327299-829cdf00-a45e-11e8-9b26-39c6a2459f0c.png)

English translation :

> Currently, the Facebook Login button does not work properly in your app. When a user clicks on it, the button must immediately lead to the connection with Facebook and the user must be able to connect to his Facebook account.

I'm afraid Facebook decided another drastic change in their validation policies...

**Comment by ApacheEx** [comment]
I hope this problem is not urgent for now, because only you've been reported.

So, I skip it for now. In case of any details, please let us know.

**Comment by ivangretsky** [comment]
Good day!
@jmartsch mentioned my report on a problem that seems similar. Could you please investigate?

**Comment by jcambien** [comment]
We have our application validated since I created this issue, without changing anything in the code. Facebook Validation team gives sometimes weird reasons for refusal, Hybridthauth is not involved in the problem.
This said, I looked at the issue description, looks like the URL that you are using is not the same before AND after the authentication process, it's prohibited by Facebook. Also, this URL should be registered in your Facebook app configuration (allowed redirections URL).
